judge/forms.py

```diff
@@ -250,8 +250,9 @@ def clean(self):
 
             credential.counter = sign_count
             credential.save(update_fields=['counter'])
-        elif self.profile.is_totp_enabled and totp_or_scratch_code:
-            if pyotp.TOTP(self.profile.totp_key).verify(totp_or_scratch_code, valid_window=self.TOLERANCE):
+        elif totp_or_scratch_code:
+            if (self.profile.is_totp_enabled and
+                    pyotp.TOTP(self.profile.totp_key).verify(totp_or_scratch_code, valid_window=self.TOLERANCE)):
```

Comment:
I think the error message farther down (L263) should probably be updated.

Reply:
Fixed.